numpy/core/overrides.py

```diff
@@ -34,6 +35,15 @@ def set_array_function_like_doc(public_api):
     return public_api
 
 
+def add_signature(function, docstring):
+    try:
+        function.__signature__ = inspect._signature_fromstr(
+            inspect.Signature, function,
+            docstring.strip().split('\n')[0])
+    except ValueError:
```

Comment:
This will need `except Exception` probably, there's other ways this could conceivably fail.